.github/workflows/quay_binaries_push.yml

```diff
@@ -5,26 +5,103 @@ on:
   push:
     tags:
       - 'v*'
+  pull_request:
+    
 
 # Check out code, podman login, run podman build, push to 
 jobs:
-  push-binaries:
+  single-arch-build:
+    name: Single-Arch Build
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Podman login
-        run: podman login -u ${{ secrets.QUAY_USER }} -p ${{ secrets.QUAY_TOKEN }} quay.io
+      - name: Install qemu dependency
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y qemu-user-static
+      
+      - name: Build Image
+        id: build_image_singlearch
+        uses: redhat-actions/buildah-build@v2
+        with:
+          image: quay.io/konveyor/oadp-cli-binaries
+          tags: ${{ github.ref_type == 'tag' && format('latest {0}', github.ref_name) || 'test' }}
+          archs: amd64
+          containerfiles: |
+            ./Containerfile.download
 
-      - name: Podman build and tag
+      - name: Echo Outputs
         run: |
-          podman build -f Dockerfile.download \
-            -t quay.io/konveyor/oadp-cli-binaries:${{ github.ref_name }} \
-            -t quay.io/konveyor/oadp-cli-binaries:latest .
+          echo "Image: ${{ steps.build_image_singlearch.outputs.image }}"
+          echo "Tags: ${{ steps.build_image_singlearch.outputs.tags }}"
+          echo "Tagged Image: ${{ steps.build_image_singlearch.outputs.image-with-tag }}"
+
+      - name: Check images created
+        run: buildah images | grep 'quay.io/konveyor/oadp-cli-binaries'
+        
+      - name: Check image metadata
+        run: |
+          set -x
+          buildah inspect ${{ steps.build_image_singlearch.outputs.image-with-tag }} | jq ".OCIv1.architecture"
+          buildah inspect ${{ steps.build_image_singlearch.outputs.image-with-tag }} | jq ".Docker.architecture"
+  
+  multi-arch-build:
+    name: Multi-Arch Build
+    runs-on: ubuntu-latest
+    needs: single-arch-build
+    steps:
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install qemu dependency
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y qemu-user-static
+      
+      - name: Build Image
+        id: build_image_multiarch
+        uses: redhat-actions/buildah-build@v2
+        with:
+          image: quay.io/konveyor/oadp-cli-binaries
+          tags: ${{ github.ref_type == 'tag' && format('latest {0}', github.ref_name) || 'test' }}
+          archs: amd64,arm64,ppc64le,s390x
```

Comment:
building for ppc and s390 will be wasted effort here

Reply:
unless p & z QE are going to install from oadp-dev, which they will not..  IMHO it's a waste.  It will be built downstream on p & z

Reply:
Just so that any oadp-operator installs on those archs don't hang/fail on the binary server pod start

Reply:
It's ok GitHub action is free 😁